File: conformance/tests/if-else/if-else.runar.zig
const runar = @import("runar");

pub const IfElse = struct {
    pub const Contract = runar.SmartContract;

    limit: i64,

    pub fn init(limit: i64) IfElse {
        return .{ .limit = limit };
    }

    pub fn check(self: *const IfElse, value: i64, mode: bool) void {
        var result: i64 = 0;
        if (mode) {
            result = value + self.limit;
        } else {
            result = value - self.limit;
        }
        runar.assert(result > 0);
    }
};

File: conformance/tests/if-else/if-else.runar.go
//go:build ignore

package contract

import "runar"

type IfElse struct {
	runar.SmartContract
	Limit runar.Int `runar:"readonly"`
}

func (c *IfElse) Check(value runar.Int, mode runar.Bool) {
	result := runar.Int(0)
	if mode {
		result = value + c.Limit
	} else {
		result = value - c.Limit
	}
	runar.Assert(result > 0)
}
